### PR TITLE
EIP1-3229 Add endpoint to return a summary of the Anonymous Elector Documents (AEDs)

### PR DESCRIPTION
**Pull request by Neil-Massey**
image

### CODE REVIEW

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 1]
```diff
@@ -446,6 +446,87 @@ paths:
...
+  '/eros/{eroId}/anonymous-elector-documents/applications/{applicationId}':
+    parameters:
+      - name: eroId
+        description: The ID of the Electoral Registration Office responsible for the application.
+        schema:
+          type: string
+        in: path
+        required: true
+      - name: applicationId
+        description: The identifier of the Anonymous Elector Document.
```
Comment (matt-wills-valtech) [inline src/main/resources/openapi/PrintAPIs.yaml:461]: Is that right, or is it the ID of the AED's originating application (aka `sourceReference`)?

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 2]
```diff
@@ -446,6 +446,87 @@ paths:
...
+        '200':
+          $ref: '#/components/responses/AnonymousElectorDocumentSummaries'
+        '404':
+          description: Not Found
+      operationId: get-anonymous-elector-document-summary-by-application-id
+      security:
+        - eroUserCognitoUserPoolAuthorizer: [ ]
+      x-amazon-apigateway-integration:
+        type: HTTP_PROXY
+        uri: '${base_uri}/eros/{eroId}/anonymous-elector-documents/applications/{applicationId}'
```
Comment (matt-wills-valtech) [inline src/main/resources/openapi/PrintAPIs.yaml:518]: To contradict myself slightly here, I think `/applications` is ok here as it's clearly in the path/context of `/anonymous-elector-documents`.

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 3]
```diff
@@ -803,6 +884,63 @@ components:
         - photoLocation
         - validOnDate
 
+    AnonymousElectorDocumentSummariesResponse:
+      title: AnonymousElectorDocumentSummariesResponse
+      type: object
+      description: A response containing a collection of Anonymous Elector Document summaries
```
Comment (matt-wills-valtech) [inline src/main/resources/openapi/PrintAPIs.yaml:890]: Will the most recent one be returned first?

File: src/main/resources/openapi/PrintAPIs.yaml (hunk 1 above)
Comment (Neil-Massey) [reply src/main/resources/openapi/PrintAPIs.yaml:461]: You're right, it is the sourceReference, and will be found using `findByGssCodeInAndSourceTypeAndSourceReference`.  

Agree, wording is wrong, the Certificates & Temporary Certificates use the following wording:
```
The identifier of the Voter Card Application.
```

So maybe, this one should be:
```
The identifier of the Anonymous Voter Card Application.
```

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 4]
```diff
@@ -803,6 +884,63 @@ components:
...
+          $ref: '#/components/schemas/AnonymousElectorDocumentStatus'
+        userId:
+          type: string
+          format: email
+          example: fred.blogs@some-domain.co.uk
+          description: UserId of the user who made the print request update
+        dateTime:
+          type: string
+          format: date-time
+          description: The timestamp in ISO-8601 UTC when the print request status update was received
```
[the remark this answers is not in the document]
Comment (Neil-Massey) [reply src/main/resources/openapi/PrintAPIs.yaml:919]: Should be the timestamp corresponding to when the status changed.

File: src/main/resources/openapi/PrintAPIs.yaml (hunk 3 above)
Comment (Neil-Massey) [reply src/main/resources/openapi/PrintAPIs.yaml]: Let's say we will do that - description updated 👍

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 5]
```diff
@@ -803,6 +884,63 @@ components:
         - photoLocation
         - validOnDate
 
+    AnonymousElectorDocumentSummariesResponse:
+      title: AnonymousElectorDocumentSummariesResponse
+      type: object
+      description: A response containing a collection of Anonymous Elector Document summaries sorted with latest changes ordered first.
```
Comment (matt-wills-valtech) [inline src/main/resources/openapi/PrintAPIs.yaml:890]: 👍

File: src/main/resources/openapi/PrintAPIs.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Neil-Massey) [reply src/main/resources/openapi/PrintAPIs.yaml]: To clarify, the endpoint returns only 1 summary record for each AED and that has its latest status and the timestamp of when that status was assigned.

**Review by matt-wills-valtech — APPROVED** [review]
LGTM, thanks for finishing this off 👍

**Pull request merged**